pkg/scalers/kafka_scaler.go

```diff
@@ -86,6 +106,45 @@ func parseKafkaMetadata(metadata map[string]string) (kafkaMetadata, error) {
 		meta.lagThreshold = t
 	}
 
+	meta.authMode = kafkaAuthModeForNone
+	if val, ok := authParams["authMode"]; ok {
+		mode := kafkaAuthMode(val)
+		if mode != kafkaAuthModeForNone && mode != kafkaAuthModeForSaslPlaintext && mode != kafkaAuthModeForSaslSSL && mode != kafkaAuthModeForSaslScramSha256 && mode != kafkaAuthModeForSaslScramSha512 {
```

Comment:
For this and for line 119 it requires the user explicitely sets `authMode` to `none`.  I would imagine it should default to `none` if the user didn't define `authMode`.  Would be good to add in logic that if not defined, then `kafkaAuthModeForNone`

Reply:
thanks @jeffhollan ,  maybe authMode to none has no practical meaning and should be deleted. To maintain better compatibility, add TriggerAuthentication  only if auth is needed.

Reply:
hi @jeffhollan , in fact my code  is the logic you requested.  
`meta.authMode = kafkaAuthModeForNone
`
this is a default value. 

so look at :
`if val, ok := authParams["authMode"]; ok {`

if the user didn't define authMode,  authMode the value not change , is the default value.

Reply:
😂 yes I missed line 109 where you set the default

Reply:
Now, the code should be merged. docs is done.